<center><img src='img/ms_logo.jpeg' height=40% width=40%></center>

<center><h1>Z-Scores and Probability: Practice Problems</h1></center>

This notebook is designed as a way to practice answering questions of probability using the Z scores and the Standard Normal Distribution.  

Try to answer the three following questions:


**1.**  The average height of students at Make School is 5'7", with a standard deviation of 2.5 inches.  What percentage of Make School students are between 5'5" and 6' tall?
<br>
<br>
**2.**  The average startup in San Francisco raises 1.5 million dollars with a standard deviation of 400,000 dollars.  Your boss wants to raise a 3 million dollar seed round.  Based only on how other startups have done, what are the odds (percentage chance) that he will be successful?
<br>
<br>
**3.**  You've decided to track the number of jokes Alan makes at Mike's expense during a given class period.  Alan averages 2.5 jokes about Mike per class, with a standard deviation of .7 jokes.  Mike has had a rough day, and will probably start crying if Alan makes more than 2 jokes about him today.  What is the percentage chance that Mike will NOT break down crying in class today?

**NOTE:** To solve these questions, you can use a lookup-table for the Z-score, or just use the stats package inside of scipy.  

To compute the cumulative area to the **left** of a Z-score, use the cumulative density function.
<br>
<center>`st.norm.cdf(z-score)`</center>
<br>

To compute the cumulative area to the **right** of a Z-score, use the survival function.
<br>
<center>`st.norm.sf(z-score)`</center>
<br>

Recall that to calculate a Z-score for a particular point, you need to **_normalize_** it by subtracting the mean from that point, and then dividing that number by the standard deviation.  

In [1]:
import scipy.stats as st
# Example use:
# A Z-score of ~1.64 == ~95% to the left, ~5% to the right
left_tail = st.norm.cdf(1.64)
right_tail = st.norm.sf(1.64)


print("Area to the left: {}".format(left_tail))
print("Area to the right: {}".format(right_tail))
print("total area under curve: {}".format(left_tail + right_tail))

Area to the left: 0.9494974165258963
Area to the right: 0.05050258347410371
total area under curve: 1.0


In [2]:
# Now, write some code to find the answers to the questions listed above! 
# (65 - 67) / 2.5 = -0.8
# (72 - 67) / 2.5 = 2.0
# st.norm.cdf(2.0) - st.norm.cdf(-0.8)
# (3 - 1.5) / 0.4 = 3.75
# st.norm.sf(3.75)
# (2 - 2.5) / 0.7 = -0.7142857142857143
st.norm.cdf(-0.7142857142857143)

0.23752526202697649

Answer 1: 0.76539446946842404



Answer 2: 8.8417285200803773e-05



Answer 3: 0.23752526202697649




Query 1: 
SELECT * FROM city
WHERE population > 100000 AND countrycode = 'USA';

Query 2:
SELECT name FROM city
WHERE population > 120000 AND countrycode = 'USA';

Query 3:
SELECT * FROM city;

Query 4:
SELECT * FROM city
WHERE id = 1661;

Query 5:
SELECT * FROM city
WHERE countrycode = 'JPN';

Query 6:
SELECT name FROM city
WHERE countrycode = 'JPN';

Query 7:
SELECT city, state FROM station

Query 8:
SELECT DISTINCT city FROM station
WHERE (id % 2) = 0;

Query 9:
SELECT (count(city) - count(DISTINCT city)) FROM station;